Exec/unit_tests/diffusion_test/prob_util.H

```diff
@@ -4,29 +4,55 @@
 AMREX_GPU_HOST_DEVICE AMREX_INLINE
 Real analytic(const Real* r, const Real time, const int coord_type) {
 
-    Real exponent;
-    if (coord_type == 2) {
-        // Handle spherical coordinates
-        exponent = 3.0_rt / 2.0_rt;
-    } else if (AMREX_SPACEDIM == 2 && coord_type == 1) {
-        // Handle cylindrical coordinates
-        exponent = 3.0_rt / 2.0_rt;
-    } else {
-        exponent = AMREX_SPACEDIM / 2.0_rt;
-    }
+    Real temp = 0.0_rt;
+
+#if AMREX_SPACEDIM == 2
+    if ((!problem::gaussian_init) && (coord_type == 2)) {
+        // For spherical geometry with azimuthal symmetry,
+        // diffusion equation has analytic form of:
+        // T(r, theta, t) = A_l j_l(kr) P_l(cos(theta)) exp(-t/t_0)
+        // where k^2 = 1/(D*t_0)
+        // Here we initialize problem to a single mode, l = 1.
 
-    Real dist2 = 0.0;
-    if (coord_type == 2) {
-        dist2 = r[0]*r[0];
-    } else {
-        for (int d = 0; d < AMREX_SPACEDIM; d++) {
-            dist2 += r[d] * r[d];
+        Real kr = std::sqrt(1.0_rt / (problem::diff_coeff * problem::t_0)) * r[0];
+        temp = problem::T1 + (problem::T2 - problem::T1) *
+            (std::sin(kr) / (kr * kr) - std::cos(kr) / kr) * std::cos(r[1]) *
+            std::exp(-1.0_rt * time / problem::t_0);
+    } else
+#endif
+    {
+        Real exponent;
+        if (coord_type == 2) {
+            // Handle spherical coordinates
+            exponent = 3.0_rt / 2.0_rt;
+        } else if (AMREX_SPACEDIM == 2 && coord_type == 1) {
+            // Handle cylindrical coordinates
+            exponent = 3.0_rt / 2.0_rt;
+        } else {
+            exponent = AMREX_SPACEDIM / 2.0_rt;
         }
-    }
 
-    Real temp = problem::T1 + (problem::T2 - problem::T1) *
-        std::pow(problem::t_0 / (time + problem::t_0), exponent) *
-        std::exp(-0.25_rt * dist2 / (problem::diff_coeff * (time + problem::t_0)));
+        Real dist2 = 0.0;
+        if (coord_type == 2) {
+            dist2 = r[0] * r[0];
+#if AMREX_SPACEDIM == 2
+            // Note r[0] = r, r[1] = theta,
+            // problem::center[0] = cylindrical r-center, problem::center[1] = z-center
```

Comment:
I think this comment is wrong -- this isn't cylindrical

Reply:
Somehow I thought using cylindrical coordinate for problem::center can be more intuitive when setting up a problem (thinking directly in r-z). But now I changed it back to spherical coordinate, see comments within the code.